lib/stdlib/test/lists_SUITE.erl

```diff
@@ -2333,6 +2334,21 @@ flatten_2(Config) when is_list(Config) ->
 flatten_2_e(Config) when is_list(Config) ->
     ok.
 
+%% Test lists:transpose/1.
+transpose(Config) when is_list(Config) ->
+    [] = lists:transpose([[], [], [], []]),
+    [[a0, b0, c0], [a1, b1, c1], [a2, b2, c2]] =
+        lists:transpose([[a0, a1, a2], [b0, b1, b2], [c0, c1, c2]]),
+    [[a0, b0], [a1, b1], [a2, b2]] =
+        lists:transpose([[a0, a1, a2], [b0, b1, b2]]),
+
```

Comment:
To thoroughly test the new implementation, here is an exhaustive test of all matrices from 1x1 up to 10x10:

```suggestion
    _ = [begin
             Matrix = [[make_ref() || _ <- lists:seq(1, M)] ||
                          _ <- lists:seq(1, N)],
             Matrix = lists:transpose(lists:transpose(Matrix))
         end || N <- lists:seq(1, 10), M <- lists:seq(1, 10)],
 
```